app/src/main/java/org/fossasia/openevent/api/processor/VersionApiProcessor.java

```diff
@@ -18,66 +24,82 @@
-                download.downloadSpeakers();
-                download.downloadTracks();
-                download.downloadMicrolocations();
-                download.downloadSession();
-                download.downloadSponsors();
-
-            } else if ((dbSingleton.getVersionIds().getId() != version.getId())) {
-                DataDownload download = new DataDownload();
-                if (dbSingleton.getVersionIds().getEventVer() != version.getEventVer()) {
-                    download.downloadEvents();
-                    Log.d(TAG, "events");
-                }
-                if (dbSingleton.getVersionIds().getSpeakerVer() != version.getSpeakerVer()) {
-                    download.downloadSpeakers();
-                    Log.d(TAG, "speaker");
-                }
-                if (dbSingleton.getVersionIds().getSponsorVer() != version.getSponsorVer()) {
-                    download.downloadSponsors();
-                    Log.d(TAG, "sponsor");
-                }
-                if (dbSingleton.getVersionIds().getTracksVer() != version.getTracksVer()) {
-                    download.downloadTracks();
-                    Log.d(TAG, "tracks");
-                }
-                if (dbSingleton.getVersionIds().getSessionVer() != version.getSessionVer()) {
-                    download.downloadSession();
-                    Log.d(TAG, "session");
+    public void success(final VersionResponseList versionResponseList, Response response) {
+        CommonTaskLoop.getInstance().post(new Runnable() {
+            @Override
+            public void run() {
+                ArrayList<String> queries = new ArrayList<>();
+                DbSingleton dbSingleton = DbSingleton.getInstance();
+                Log.d("!", "3");
+                for (Version version : versionResponseList.versions) {
+
+                    if ((dbSingleton.getVersionIds() == null)) {
+                        queries.add(version.generateSql());
+                        dbSingleton.insertQueries(queries);
+                        DataDownload download = new DataDownload();
+                        download.downloadEvents();
+                        download.downloadSpeakers();
+                        download.downloadTracks();
+                        download.downloadMicrolocations();
+                        download.downloadSession();
+                        download.downloadSponsors();
+
+
+                    } else if ((dbSingleton.getVersionIds().getId() != version.getId())) {
+                        DataDownload download = new DataDownload();
+                        if (dbSingleton.getVersionIds().getEventVer() != version.getEventVer()) {
+                            download.downloadEvents();
+                            Log.d(TAG, "events");
+
+                        }
+                        if (dbSingleton.getVersionIds().getSpeakerVer() != version.getSpeakerVer()) {
+                            download.downloadSpeakers();
```

Comment:
These are asynchronous operations. You dunno the status of these calls, unless onError/onSuccess is called.